packages/components/src/input-control/stories/index.story.tsx

```diff
@@ -82,3 +89,32 @@ WithEdgeLabel.args = {
 	__unstableInputWidth: '20em',
 	labelPosition: 'edge',
 };
+
+export const ShowPassword: StoryFn< typeof InputControl > = ( args ) => {
+	const [ visible, toggleVisible ] = useReducer(
+		( value ) => ! value,
+		false
+	);
+	return (
+		<InputControl
+			type={ visible ? 'text' : 'password' }
+			label="Password"
+			suffix={
+				<InputControlSuffixWrapper>
+					<Tooltip
+						text={ visible ? 'Hide password' : 'Show password' }
+					>
+						<Button
+							icon={ visible ? unseen : seen }
+							onClick={ toggleVisible }
+							aria-label={
+								visible ? 'Hide password' : 'Show password'
+							}
```

Comment:
The tooltip functionality is actually built into Button. It will set both a tooltip and the aria-label if you do this:

```suggestion
							label={
								visible ? 'Hide password' : 'Show password'
							}
```

Reply:
Ok, I will go with that. Off-topic but I'm not sure I like bundling components like this, though I understand the reasoning behind it.

Reply:
Yes, the Button component is ancient and we'd probably design a very different API if we started from scratch today.